backend/molgenis-emx2-datamodels/src/main/java/org/molgenis/emx2/datamodels/profiles/SchemaFromProfile.java

```diff
@@ -59,23 +65,132 @@ public SchemaMetadata create() throws MolgenisException {
   /** From a classpath dir, get all EMX2 model files and slice for profiles */
   public List<Row> getProfilesFromAllModels(String directory)
       throws URISyntaxException, IOException {
+
+    // first, load everything into maps
+    Map<String, Row> tableDeclarationsByTableName = new HashMap<>();
+    Map<String, List<Row>> columnDeclarationsByTableName = new HashMap<>();
     List<Row> keepRows = new ArrayList<>();
     String[] modelsList = getResourceListing(directory);
-    for (String schemaLoc : modelsList) {
 
+    for (String schemaLoc : modelsList) {
       Iterable<Row> rowIterable =
           CsvTableReader.read(
               new InputStreamReader(
                   Objects.requireNonNull(
                       getClass().getResourceAsStream(directory + File.separator + schemaLoc))));
-
       for (Row row : rowIterable) {
-        List<String> profiles = csvStringToList(row.getString("profiles"));
-        if (profiles.isEmpty()) {
-          throw new MolgenisException("No profiles for " + row);
+
+        String tableName = row.getString("tableName");
```

Comment:
You could have done this after parsing a 'schema' (you could have used https://github.com/molgenis/molgenis-emx2/blob/5ace029d142ff3496f9af6b7e25e79ed1fdad8d4/backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/emx2/Emx2.java#L42)